internal/config/validation.go

```diff
@@ -281,3 +297,61 @@ func getPlaceholderEntries(m Measurements) []uint32 {
 
 	return placeholders
 }
+
+func validateK8sVersion(fl validator.FieldLevel) bool {
+	return versions.IsSupportedK8sVersion(fl.Field().String())
+}
+
+func registerVersionCompatibilityError(ut ut.Translator) error {
+	return ut.Add("version_compatibility", "{0} specifies an invalid version: {1}", true)
+}
+
+func translateVersionCompatibilityError(ut ut.Translator, fe validator.FieldError) string {
+	err := validateVersionCompatibilityHelper(fe.Field(), fe.Value().(string))
+	var msg string
+
+	switch err {
+	case compatibility.ErrSemVer:
+		msg = fmt.Sprintf("configured version (%s) does not adhere to SemVer syntax", fe.Value().(string))
+	case compatibility.ErrMajorMismatch:
+		msg = fmt.Sprintf("the CLI's major version (%s) has to match your configured major version (%s)", constants.VersionInfo, fe.Value().(string))
+	case compatibility.ErrMinorDrift:
+		msg = fmt.Sprintf("only the CLI (%s) can be up to one minor version newer than the configured version (%s)", constants.VersionInfo, fe.Value().(string))
```

Comment:
```suggestion
	switch {
	case errors.Is(compatibility.ErrSemVer):
		msg = fmt.Sprintf("configured version (%s) does not adhere to SemVer syntax", fe.Value().(string))
	case errors.Is(compatibility.ErrMajorMismatch):
		msg = fmt.Sprintf("the CLI's major version (%s) has to match your configured major version (%s)", constants.VersionInfo, fe.Value().(string))
	case errors.Is(compatibility.ErrMinorDrift):
		msg = fmt.Sprintf("only the CLI (%s) can be up to one minor version newer than the configured version (%s)", constants.VersionInfo, fe.Value().(string))
```

Using `switch err{ case ErrType: }` does not catch wrapped errors, since its basically the same as `if err == ErrType`